shared/readline/readline.c

```diff
@@ -293,16 +293,16 @@ int readline_process_char(int c) {
                 break;
             #if MICROPY_REPL_EMACS_WORDS_MOVE
             case 'b':
-#if MICROPY_REPL_EMACS_EXTRA_WORDS_MOVE
-backward_word:
-#endif
+                #if MICROPY_REPL_EMACS_EXTRA_WORDS_MOVE
+            backward_word:
+                #endif
```

Comment:
???

Reply:
The following addition to the `fixup_c` function catches these edge cases, but I don't love it-

```patch
diff --git a/tools/codeformat.py b/tools/codeformat.py
index e2e5c0cc0..97b466d06 100755
--- a/tools/codeformat.py
+++ b/tools/codeformat.py
@@ -109,8 +109,13 @@ def fixup_c(filename):
                 directive = m.group(2)
                 if directive in ("if ", "ifdef ", "ifndef "):
                     l_next = lines[0]
+                    label = re.match(f"^ +[a-z_]+:", l_next)
                     indent_next = len(re.match(r"( *)", l_next).group(1))
-                    if indent - 4 == indent_next and re.match(r" +(} else |case )", l_next):
+                    if label:
+                        l = l[indent:]
+                        lines[0] = lines[0].lstrip()
+                        dedent_stack.append(0)
+                    elif indent - 4 == indent_next and re.match(r" +(} else |case )", l_next):
                         # This #-line (and all associated ones) needs dedenting by 4 spaces.
                         l = l[4:]
                         dedent_stack.append(indent - 4)
```

Reply:
What about the slightly more verbose:

```c
#if MICROPY_REPL_EMACS_EXTRA_WORDS_MOVE
            case 'b':
            backward_word:
#elif MICROPY_REPL_EMACS_WORDS_MOVE
            case 'b':
#endif
```

... to avoid nesting multiple #ifdefs in these?

Reply:
Or, as `forward_word` and `backward_word` are labels, maybe they could always be included but marked as unused to avoid the warning (not sure how to do that for labels, though...)

Reply:
I don't know what level of crime this is, but it seems to work and doesn't rely on any compiler-specific pragmas:

```c
            #if MICROPY_REPL_EMACS_WORDS_MOVE
            case 'b':
                goto backward_word; // Ensure label always used
backward_word:
                redraw_step_back = cursor_count_word(0);
                rl.escape_seq = ESEQ_NONE;
                break;
            case 'f':
                goto forward_word; // Ensure label is always used
forward_word:
                redraw_step_forward = cursor_count_word(1);
                rl.escape_seq = ESEQ_NONE;
                break;
            case 'd':
                vstr_cut_out_bytes(rl.line, rl.cursor_pos, cursor_count_word(1));
                redraw_from_cursor = true;
                rl.escape_seq = ESEQ_NONE;
                break;
            case 127:
                goto backward_kill_word; // Ensure label is always used
backward_kill_word:
                redraw_step_back = cursor_count_word(0);
                vstr_cut_out_bytes(rl.line, rl.cursor_pos - redraw_step_back, redraw_step_back);
                redraw_from_cursor = true;
                rl.escape_seq = ESEQ_NONE;
                break;
            #endif
```